app/src/pages/inside/dashboardPage/widgets/projectActivity/activities/analysisProperties.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import Parser from 'html-react-parser';
+import classNames from 'classnames/bind';
+import { injectIntl, intlShape, defineMessages, FormattedMessage } from 'react-intl';
+import { ANALYSIS } from 'common/constants/settingsTabs';
+import styles from './common.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  auto_analyze: {
+    id: 'UpdateAnalysisSettings.autoAnalyze',
+    defaultMessage: 'switch Auto Analysis',
+  },
+  number_of_log_lines: {
+    id: 'UpdateAnalysisSettings.numberOfLogLines',
+    defaultMessage: 'Number of log lines',
+  },
+  min_doc_freq: {
+    id: 'UpdateAnalysisSettings.minDocFreq',
+    defaultMessage: 'Minimum document frequency',
+  },
+  min_should_match: {
+    id: 'UpdateAnalysisSettings.minShouldMatch',
+    defaultMessage: 'Minimum should match',
+  },
+  min_term_freq: {
+    id: 'UpdateAnalysisSettings.minTermFreq',
+    defaultMessage: 'Minimum term frequency',
+  },
+  analyze_mode: {
+    id: 'UpdateAnalysisSettings.analyzeMode',
+    defaultMessage: 'Base for Auto Analysis',
+  },
+  from: {
+    id: 'AnalysisProperties.from',
+    defaultMessage: 'from',
+  },
+  to: {
+    id: 'AnalysisProperties.to',
+    defaultMessage: 'to',
+  },
+  all: {
+    id: 'AnalysisProperties.all',
+    defaultMessage: 'All',
+  },
+  on: {
+    id: 'AnalysisProperties.on',
+    defaultMessage: 'ON',
+  },
+  off: {
+    id: 'AnalysisProperties.off',
+    defaultMessage: 'OFF',
+  },
+});
+
+@injectIntl
+export class AnalysisProperties extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    activity: PropTypes.object,
+  };
+  static defaultProps = {
+    activity: {},
+  };
+
+  getActivityHistory = (activity) => {
+    const size = Object.keys(activity.history).length;
+    let i = 0;
+    let activities = '';
+    Object.keys(activity.history).forEach((key) => {
+      i += 1;
+      if (activity.history[key].newValue && activity.history[key].oldValue) {
+        const activityName = Parser(this.props.intl.formatMessage(messages[key]));
+        const from = Parser(this.props.intl.formatMessage(messages.from));
+        const oldValue =
+          this.valueReplacer(activity.history[key].oldValue) || activity.history[key].oldValue;
+        const to = Parser(this.props.intl.formatMessage(messages.to));
+        const newValue =
+          this.valueReplacer(activity.history[key].newValue) || activity.history[key].newValue;
+        const end = i < size ? ', ' : '.';
```

Comment:
You can create array of activities and just `join` it at the end.

For example `['String 1', 'String 2', 'String 3'].join(', ') + '.'` will return string `String 1, String 2, String 3.`
I guess it approach will be much simpler.

Reply:
Fixed